frontend/rolecall/src/app/performance/performance-editor.component.ts

```diff
@@ -297,53 +351,36 @@ export class PerformanceEditor implements OnInit, OnDestroy, AfterViewChecked {
 
   // Step 1 -------------------------------------------------------
 
-  allPerformances: Performance[] = [];
-  selectedPerformance: Performance;
-  dateStr: string;
-  date: Date;
-
   onSelectRecentPerformance(perf: Performance) {
     this.selectedPerformance = perf;
     this.updateDateString();
   }
 
   updateDateString() {
-    let timeZoneOffset = new Date().getTimezoneOffset() * 60000;
+    const timeZoneOffset = new Date().getTimezoneOffset() * 60000;
     this.date = new Date(this.state.step_1.date - timeZoneOffset);
-    let iso = this.date.toISOString();
-    let isoSplits = iso.slice(0, iso.length - 1).split(":");
-    this.dateStr = isoSplits[0] + ":" + isoSplits[1];
+    const iso = this.date.toISOString();
+    const isoSplits = iso.slice(0, iso.length - 1).split(':');
+    this.dateStr = isoSplits[0] + ':' + isoSplits[1];
   }
 
   onStep1Input(field: string, value: any) {
-    if (field == "title") {
-      let val = value as InputEvent;
-      this.state.step_1.title = val.target['value'];
-    }
-    else if (field == "city") {
-      let val2 = value as InputEvent;
-      this.state.step_1.city = val2.target['value'];
-    }
-    else if (field == "country") {
-      let val2 = value as InputEvent;
-      this.state.step_1.country = val2.target['value'];
-    }
-    else if (field == "state") {
-      let val2 = value as InputEvent;
-      this.state.step_1.state = val2.target['value'];
-    }
-    else if (field == "venue") {
-      let val2 = value as InputEvent;
-      this.state.step_1.venue = val2.target['value'];
-    }
-    else if (field == "date") {
-      let val3 = value as InputEvent;
-      this.state.step_1.date = Date.parse(val3.target['value'])
+    const val = value as InputEvent;
+    if (field === 'title') {
+      this.state.step_1.title = (val.target as any).value;
```

Comment:
The actual type of "val.target" looks like is "EventTarget | null", so let's completely avoid adding the `any` type as it could hide bugs. For example, how does it impact our code if the `target` property is actually null?